pkg/controller/common/statefulset/reconcile.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package statefulset
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/hash"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/maps"
+)
+
+// Params to specify a StatefulSet specification.
+type Params struct {
+	Name                 string
+	Namespace            string
+	ServiceName          string
+	Selector             map[string]string
+	Labels               map[string]string
+	PodTemplateSpec      corev1.PodTemplateSpec
+	VolumeClaimTemplates []corev1.PersistentVolumeClaim
+	Replicas             int32
+	RevisionHistoryLimit *int32
+}
+
+// New creates a StatefulSet from the given params.
+func New(params Params) appsv1.StatefulSet {
+	return appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      params.Name,
+			Namespace: params.Namespace,
+			Labels:    params.Labels,
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: &params.Replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: params.Selector,
+			},
+			Template:             params.PodTemplateSpec,
+			VolumeClaimTemplates: params.VolumeClaimTemplates,
+			ServiceName:          params.ServiceName,
+			RevisionHistoryLimit: params.RevisionHistoryLimit,
+			// always RollingUpdate strategy as OnDelete would require the user
+			// to delete pods and this is not fully with the operator reconcile logic.
+			UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+				Type:          appsv1.RollingUpdateStatefulSetStrategyType,
+				RollingUpdate: nil,
+			},
+			// default statefulSet behaviour
+			PodManagementPolicy: appsv1.OrderedReadyPodManagement,
```

Comment:
thanks for the comment @barkbay, tbh I went with the `OrderedReadyPodManagement` as it is the default one for `statefulset` and because the latter is mostly used for the constant pod names and the ordered pod management so you can update/rollback without downtime between versions. But I see your point, maybe have that as an option in the spec?

Reply:
My gut feeling is that unless it is necessary to start the Pods in a specific order I would go for `Parallel`, but I have to admit that I have no other argument than that this would allow Pods to start more quickly. 😄 

> But I see your point, maybe have that as an option in the spec?

👍

Reply:
@barkbay I pushed an update, now `PodManagementPolicy` defaults to `Parallel` but can be specified through yaml to `OrderedReadyPodManagement`